### ISSUE TITLE
java.lang.IllegalStateException: Should not be called in IJ 2023.3

### ISSUE THREAD

**Issue by valb3r** [GDNext-GmbH/polybpmn#11]
**Describe the bug**
In IntelliJ 2023.3
```
java.lang.IllegalStateException: Should not be called
	at com.intellij.openapi.fileEditor.AsyncFileEditorProvider.createEditorAsync(AsyncFileEditorProvider.kt:20)
```
When trying to open BPMN file

**Comment by benius** [comment]
We got the same error message when trying to open [a BPMN 2.0 file](https://drive.google.com/file/d/1LxgBRg0DKRXpCmdfChL7N9ij219oPoc7/view?usp=sharing) on IDEA.

Here is the version description of installed IDEA:
```
IntelliJ IDEA 2023.3 (Ultimate Edition)
Build #IU-233.11799.241, built on December 2, 2023
Runtime version: 17.0.9+7-b1087.7 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 14.1.2
GC: G1 Young Generation, G1 Old Generation
Memory: 10240M
Cores: 12
Metal Rendering is ON
Registry:
  debugger.new.tool.window.layout=true
  ide.experimental.ui=true
  editor.minimap.enabled=true
Non-Bundled Plugins:
  com.jetbrains.plugins.ini4idea (233.11799.244)
  net.sjrx.intellij.plugins.systemdunitfiles (223.230322.126)
  com.intellij.plugins.macoskeymap (232.6095.10)
  com.intellij.properties.bundle.editor (233.11799.172)
  Key Promoter X (2023.3.0)
  ink.organics.pojo2json (2.0.3)
  com.intellij.spring.websocket (233.11799.196)
  com.intellij.spring.batch (233.11799.196)
  Dart (233.11799.172)
  polybpmn-intellij-visualizer (1.2.3)
  com.intellij.bigdatatools.core (233.11799.267)
  com.intellij.bigdatatools.binary.files (233.11799.267)
  com.intellij.bigdatatools.rfs (233.11799.267)
  com.intellij.bigdatatools.metastore.core (233.11799.267)
  com.intellij.bigdatatools.flink (233.11799.267)
  org.jetbrains.plugins.docker.gateway (233.11799.244)
  com.intellij.bigdatatools.kafka (233.11799.267)
  com.intellij.bigdatatools.spark (233.11799.267)
  com.intellij.bigdatatools.zeppelin (233.11799.267)
  com.intellij.bigdatatools (233.11799.267)
Kotlin: 233.11799.241-IJ

```

Here's the detailed errors when opening the BPMN file:
```
java.lang.IllegalStateException: Should not be called
	at com.intellij.openapi.fileEditor.AsyncFileEditorProvider.createEditorAsync(AsyncFileEditorProvider.kt:20)
	at com.gdnext.bpmn.intellij.plugin.core.c.a.b$a.a(Unknown Source)
	at com.gdnext.bpmn.intellij.plugin.core.c.a.b.createEditorAsync(Unknown Source)
	at com.intellij.openapi.fileEditor.AsyncFileEditorProvider$createEditorBuilder$2.invoke(AsyncFileEditorProvider.kt:25)
	at com.intellij.openapi.fileEditor.AsyncFileEditorProvider$createEditorBuilder$2.invoke(AsyncFileEditorProvider.kt:25)
	at com.intellij.openapi.application.rw.InternalReadAction.insideReadAction(InternalReadAction.kt:108)
	at com.intellij.openapi.application.rw.InternalReadAction.access$insideReadAction(InternalReadAction.kt:16)
	at com.intellij.openapi.application.rw.InternalReadAction$tryReadCancellable$2.invoke(InternalReadAction.kt:95)
	at com.intellij.openapi.application.rw.InternalReadAction$tryReadCancellable$2.invoke(InternalReadAction.kt:94)
	at com.intellij.openapi.application.rw.CancellableReadActionKt$cancellableReadActionInternal$1.invoke$lambda$1$lambda$0(cancellableReadAction.kt:38)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1075)
	at com.intellij.openapi.application.rw.CancellableReadActionKt$cancellableReadActionInternal$1.invoke$lambda$1(cancellableReadAction.kt:36)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtilService.runActionAndCancelBeforeWrite(ProgressIndicatorUtilService.java:73)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:128)
	at com.intellij.openapi.application.rw.CancellableReadActionKt$cancellableReadActionInternal$1.invoke(cancellableReadAction.kt:34)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContextInner(coroutines.kt:321)
	at com.intellij.openapi.progress.CoroutinesKt.blockingContext(coroutines.kt:310)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal(cancellableReadAction.kt:31)
	at com.intellij.openapi.application.rw.InternalReadAction.tryReadCancellable(InternalReadAction.kt:94)
	at com.intellij.openapi.application.rw.InternalReadAction.tryReadAction(InternalReadAction.kt:78)
	at com.intellij.openapi.application.rw.InternalReadAction.readLoop(InternalReadAction.kt:65)
	at com.intellij.openapi.application.rw.InternalReadAction.access$readLoop(InternalReadAction.kt:16)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$4.invokeSuspend(InternalReadAction.kt:44)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$4.invoke(InternalReadAction.kt)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$4.invoke(InternalReadAction.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:78)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:167)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at com.intellij.openapi.application.rw.InternalReadAction.runReadAction(InternalReadAction.kt:40)
	at com.intellij.openapi.application.rw.PlatformReadWriteActionSupport.executeReadAction(PlatformReadWriteActionSupport.kt:38)
	at com.intellij.openapi.application.ReadWriteActionSupport.executeReadAction$default(ReadWriteActionSupport.kt:15)
	at com.intellij.openapi.application.CoroutinesKt.constrainedReadAction(coroutines.kt:58)
	at com.intellij.openapi.application.CoroutinesKt.readAction(coroutines.kt:25)
	at com.intellij.openapi.fileEditor.AsyncFileEditorProvider.createEditorBuilder$suspendImpl(AsyncFileEditorProvider.kt:25)
	at com.intellij.openapi.fileEditor.AsyncFileEditorProvider.createEditorBuilder(AsyncFileEditorProvider.kt)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImplKt$createBuilders$2$1$1.invokeSuspend(FileEditorManagerImpl.kt:2403)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)

```

**Comment by valb3r** [comment]
@benius Thanks for feedback, 
the issue is only in IJ version 2023.3, threading model has changed there. So if it is OK for you, you can open in the file in IJ 2023.2.5 while we are fixing the bug

**Comment by valb3r** [comment]
Fixed in v1.2.4
